File: CoreAngularDemo.API/CoreAngularDemo.DAL/Models/Entities/IssueLog.cs
using System;
using System.Collections.Generic;
using CoreAngularDemo.DAL.Models.Entities.Abstractions;

namespace CoreAngularDemo.DAL.Models.Entities
{
    public class IssueLog : IAuditableEntity, IBaseEntity
    {
        public IssueLog()
        {
            Document = new HashSet<Document>();
            IssueLogParts = new HashSet<IssueLogPart>();
        }

        public int Id { get; set; }
        public string Description { get; set; }
        public decimal? Expenses { get; set; }
        public int? OldStateId { get; set; }
        public int? NewStateId { get; set; }
        public int? SupplierId { get; set; }
        public int? ActionTypeId { get; set; }
        public int? IssueId { get; set; }
        public int? WorkTypeId { get; set; }
        public DateTime? CreatedDate { get; set; }
        public DateTime? UpdatedDate { get; set; }
        public string CreatedById { get; set; }
        public string UpdatedById { get; set; }

        public virtual ActionType ActionType { get; set; }
        public virtual User Create { get; set; }
        public virtual Issue Issue { get; set; }
        public virtual User Mod { get; set; }
        public virtual State NewState { get; set; }
        public virtual State OldState { get; set; }
        public virtual Supplier Supplier { get; set; }
        public virtual WorkType WorkType { get; set; }

        public virtual ICollection<Document> Document { get; set; }
        public virtual ICollection<IssueLogPart> IssueLogParts { get; set; }
    }
}

File: CoreAngularDemo.Client/src/app/modules/shared/models/issuelog.ts
import { Issue } from './issue';
import { State } from './state';
import { ActionType } from './action-type';
import { Supplier } from './supplier';
import { Document } from './document';
import { TEntity } from '../../core/models/entity/entity';
import { WorkType } from './work-type';

export class IssueLog extends TEntity<IssueLog> {
  description: string;
  expenses: number;
  actionType: ActionType;
  workType: WorkType;
  issue: Issue;
  newState: State;
  oldState: State;
  supplier: Supplier;
  updatedDate: Date;
  createdDate: Date;
  documents: Array<Document>;
  actionTypeName: string;
  workTypeName: string;
  issueName: string;
  newStateName: string;
  oldStateName: string;
  supplierName: string;
  
  constructor(issueLog: Partial<IssueLog>) {
    super(issueLog);
    this.oldState = new State(this.oldState);
    this.newState = new State(this.newState);
    this.actionType = new ActionType(this.actionType);
    this.supplier = new Supplier(this.supplier);
    this.workType = new WorkType(this.workType);
    this.createdDate = new Date(this.createdDate);
    this.updatedDate = new Date(this.updatedDate);
  }
}
